### PR TITLE
develop: refactor vocabulary structure and minor fixes

### PR DESCRIPTION
**Pull request by ppanero**
- closes https://github.com/inveniosoftware/docs-invenio-rdm/issues/199#event-5111507325

How the menu's look like:
Screenshot 2021-08-05 at 11 26 20

Screenshot 2021-08-05 at 11 26 27

Screenshot 2021-08-05 at 11 26 40

### CODE REVIEW

File: docs/develop/example_subjects_extension.md [hunk 1]
```diff
@@ -1,148 +0,0 @@
-# Example Subjects Extension
```
Comment (ppanero) [inline docs/develop/example_subjects_extension.md:1]: I feel a bit bad about removing this... but it does not match the vocabularies section nor much the extensions one as it is now... However, I think we should keep it somehow somewhere... open to ideas :)

File: docs/customize/vocabularies/subjects.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Subjects (MeSH)
```
Comment (ppanero) [inline docs/customize/vocabularies/subjects.md:1]: @fenekku can you crosscheck Im not saying anything completly wrong in this section 👼

File: docs/develop/example_subjects_extension.md (hunk 1 above)
Comment (fenekku) [reply docs/develop/example_subjects_extension.md:1]: When we bring the extensions section back, I will bring it back.

File: docs/customize/vocabularies/subjects.md (hunk 2 above)
Comment (fenekku) [reply docs/customize/vocabularies/subjects.md:1]: I will just explain how to add a custom subjects here and point to the MeSH one as an example.

**Pull request merged**